ui/components/src/Legend/legend-model.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 The Perses Authors
```

Comment:
Eunice recommended just naming this `model`, which I like, but isn't aligned with the `component-name-model` I've seen in a few other places. I think it might be less confusing to fix the naming of those in a separate refactor PR.